tests/isolated/patcher_threadpoolexecutor.py

```diff
@@ -0,0 +1,18 @@
+# Issue #508: test ThreadPoolExecutor with monkey-patching
+__test__ = False
+
+if __name__ == '__main__':
+    import eventlet
+    eventlet.monkey_patch()
+
+    import sys
+
+    # Futures is only included in 3.2 or later
+    if sys.version_info >= (3, 2):
+        from concurrent import futures
+
+        with futures.ThreadPoolExecutor(max_workers=1) as executor:
+            future = executor.submit(pow, 2, 3)
+            res = future.result()
+        assert res == 8, '2^3 should be 8, not %s' % res
+    print('pass')
```

Comment:
This feels like a bit of an indirect test; something like
```
import eventlet
eventlet.monkey_patch()
import threading
try:
    import queue
except ImportError:
    import Queue as queue  # py2

def tgt(q):
    print(q.get())

# SimpleQueue is sufficient, but was only added in 3.7
q = getattr(queue, 'SimpleQueue', queue.Queue)()
t = threading.Thread(target=tgt, args=(q,))
t.start()
q.put('pass')
t.join()
```
seems a little more obvious -- could even tighten it to just use `eventlet.monkey_patch(all=False, thread=True)`.

OTOH, this more-clearly demonstrates the fact that we had a thing that was working when monkey-patched prior to 3.7 and stopped working after. \*shrug\*